File: Parcialito 1/División y Conquista/C/quick_sort.c
#include <stdio.h>
#include "vector.h"

int quickSort(int arreglo[])
{
    int largo = sizeof(arreglo) / sizeof(arreglo[0]);
    
    if (largo <= 1) //caso base
        return arreglo;

    int pivot = arreglo[0];
    int mayores[] = {};
    int menores[] = {};

    for (size_t i = 0; i < largo; i++)
    {
        if (arreglo[i] > pivot)
            mayores[i] = arreglo[i];
        else
            menores [i] = arreglo[i];
    }

    return quickSort(mayores) + pivot + quickSort(menores);
}

int main()
{
    vector_t *vector = vector_crear(15);
    int arreglo[] = {4,10,8,7,6,5,3,12,14,2};

    for (size_t i = 0; i < 10; i++)
        vector_guardar(arreglo[i]);
    

    printf("%d", quickSort(arreglo));
    
    return 0;
}

File: Parcialito 1/División y Conquista/Python/quick_sort.py
def quick_sort(arreglo):
    
    if len(arreglo) <= 1: #caso base
        return arreglo

    pivot = [arreglo.pop(0)]
    mayores = []
    menores = []

    for i in range(0, len(arreglo)):
        if arreglo[i] > pivot[0]:
            mayores.append(arreglo[i])
        else:
            menores.append(arreglo[i])

    return quick_sort(menores) + pivot + quick_sort(mayores)

def main():
    arreglo = [4,10,8,7,6,5,3,12,14,2]
    arreglo = quick_sort(arreglo)
    print(arreglo)

if __name__ == "__main__":
    main()